exporter/tanzuobservabilityexporter/metrics.go

```diff
@@ -263,3 +270,200 @@ func (g *gaugeConsumer) pushSingleNumberDataPoint(
+	return r.noAggregationTemporality.Get()
+}
+
+// LogMalformed logs seeing one malformed data point.
+func (r *histogramReporting) LogMalformed(metric pdata.Metric) {
+	namef := zap.String(metricNameString, metric.Name())
+	r.logger.Debug("Malformed histogram", namef)
+	r.malformedHistograms.Inc()
+}
+
+// LogLeTagInUse logs seeing one data point using the 'le' tag
+func (r *histogramReporting) LogLeTagInUse(metric pdata.Metric) {
+	namef := zap.String(metricNameString, metric.Name())
+	r.logger.Debug("le tag already in use", namef)
+	r.leInUse.Inc()
+}
+
+// LogNoAggregationTemporality logs seeing a histogram metric with no aggregation temporality
+func (r *histogramReporting) LogNoAggregationTemporality(metric pdata.Metric) {
+	namef := zap.String(metricNameString, metric.Name())
+	r.logger.Debug("histogram metric missing aggregation temporality", namef)
+	r.noAggregationTemporality.Inc()
+}
+
+// Report sends the counts in this instancce to wavefront.
+// sender is what sends to wavefront. Any errors sending get added to errs.
+func (r *histogramReporting) Report(sender gaugeSender, errs *[]error) {
+	r.malformedHistograms.Report(malformedHistogramMetricName, nil, sender, errs)
+	r.leInUse.Report(leInUseMetricName, nil, sender, errs)
+	r.noAggregationTemporality.Report(
+		noAggregationTemporalityMetricName,
+		typeIsHistogramTags,
+		sender,
+		errs)
+}
+
+type histogramConsumer struct {
+	cumulative            histogramDataPointConsumer
+	delta                 histogramDataPointConsumer
+	sender                gaugeSender
+	reporting             *histogramReporting
+	reportInternalMetrics bool
+}
+
+// newHistogramConsumer returns a metricConsumer that consumes histograms.
+// cumulative and delta handle cumulative and delta histograms respectively.
+// sender sends internal metrics to wavefront. Caller can pass nil for
+// options to get the defaults.
+func newHistogramConsumer(
+	cumulative, delta histogramDataPointConsumer,
+	sender gaugeSender,
+	options *consumerOptions,
+) typedMetricConsumer {
+	var fixedOptions consumerOptions
+	if options != nil {
+		fixedOptions = *options
+	}
+	fixedOptions.replaceZeroFieldsWithDefaults()
+	return &histogramConsumer{
+		cumulative:            cumulative,
+		delta:                 delta,
+		sender:                sender,
+		reporting:             newHistogramReporting(fixedOptions.Logger),
+		reportInternalMetrics: fixedOptions.ReportInternalMetrics,
+	}
+}
+
+func (h *histogramConsumer) Type() pdata.MetricDataType {
+	return pdata.MetricDataTypeHistogram
+}
+
+func (h *histogramConsumer) Consume(metric pdata.Metric, errs *[]error) {
+	histogram := metric.Histogram()
+	aggregationTemporality := histogram.AggregationTemporality()
+	var consumer histogramDataPointConsumer
+	switch aggregationTemporality {
+	case pdata.MetricAggregationTemporalityDelta:
+		if h.delta == nil {
+			*errs = append(*errs, errors.New("delta histograms not supported"))
+			return
+		}
```

Comment:
Replace conditional with `// TODO: follow-on pull request will add support for Delta Histograms`. Fine to propagate error for now until PR is submitted.

Reply:
Done.